pkg/executables/helm.go

```diff
@@ -18,54 +18,45 @@ const (
 )
 
 type Helm struct {
-	executable     Executable
-	registryMirror *registrymirror.RegistryMirror
-	env            map[string]string
-	insecure       bool
+	executable Executable
+	helmConfig *helm.Config // Embed HelmOptions in Helm struct
+	env        map[string]string
 }
 
-type HelmOpt func(*Helm)
-
-// WithRegistryMirror sets up registry mirror for helm.
-func WithRegistryMirror(mirror *registrymirror.RegistryMirror) HelmOpt {
-	return func(h *Helm) {
-		h.registryMirror = mirror
+// NewHelm returns a new Helm executable client.
+func NewHelm(executable Executable, opts ...helm.Opt) helm.Client {
+	helmConfig := &helm.Config{
+		Insecure: false,
 	}
-}
-
-// WithInsecure configures helm to skip validating TLS certificates when
-// communicating with the Kubernetes API.
-func WithInsecure() HelmOpt {
-	return func(h *Helm) {
-		h.insecure = true
+	for _, o := range opts {
+		o(helmConfig)
 	}
-}
 
-// join the default and the provided maps together.
-func WithEnv(env map[string]string) HelmOpt {
-	return func(h *Helm) {
-		for k, v := range env {
-			h.env[k] = v
-		}
+	defaultEnv := map[string]string{
+		"HELM_EXPERIMENTAL_OCI": "1",
 	}
-}
-
-func NewHelm(executable Executable, opts ...HelmOpt) *Helm {
 	h := &Helm{
 		executable: executable,
-		env: map[string]string{
-			"HELM_EXPERIMENTAL_OCI": "1",
-		},
-		insecure: false,
-	}
-
-	for _, o := range opts {
-		o(h)
+		helmConfig: helmConfig,
+		env:        joinEnv(defaultEnv, helmConfig.ProxyConfig),
 	}
 
 	return h
 }
 
+// joinEnv joins the default and the provided maps together, then return the
+// new map.
+func joinEnv(oldEnv, newEnv map[string]string) map[string]string {
+	env := make(map[string]string)
```

Comment:
nit: why do we need a new map? could we reuse one of them and have a "mergeMap"?

Reply:
I think it's better to avoid modifying the inputs directly here and instead return a new one. I think this method does what a `mergeMap` would do.

Reply:
not sure what the benefit is if you end map discarding both inputs in the caller
but ok, then at least allocate the memory in advance?

Reply:
Okay, I've considered it and after thinking about the additional code required to allocate the memory in advance, I decided to go with the original suggestion. Previously, I was mainly concerned that someone using this could have unexpected results due to the side effects on the input. Since this an internal function, there doesn't seem to be that risk.